### ISSUE TITLE
Completion windows donot resize properly for certain Rust types

### ISSUE THREAD

**Issue by salimp2009** [hrsh7th/nvim-cmp#1673]
### FAQ

- [X] I have checked the [FAQ](https://github.com/hrsh7th/nvim-cmp/blob/main/doc/cmp.txt) and it didn't resolve my problem.

### Announcement

- [X] I have checked [Breaking change announcement](https://github.com/hrsh7th/nvim-cmp/issues/231).

### Minimal reproducible full config

```vim
if has('vim_starting')
  set encoding=utf-8
endif
scriptencoding utf-8

if &compatible
  set nocompatible
endif

let s:plug_dir = expand('/tmp/plugged/vim-plug')
if !filereadable(s:plug_dir .. '/plug.vim')
  execute printf('!curl -fLo %s/autoload/plug.vim --create-dirs https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim', s:plug_dir)
end

execute 'set runtimepath+=' . s:plug_dir
call plug#begin(s:plug_dir)
Plug 'hrsh7th/nvim-cmp'
Plug 'hrsh7th/cmp-buffer'
Plug 'hrsh7th/cmp-nvim-lsp'
Plug 'hrsh7th/vim-vsnip'
Plug 'neovim/nvim-lspconfig'
Plug "simrat39/rust-tools.nvim" 
call plug#end()
PlugInstall | quit

" Setup global configuration. More on configuration below.
lua << EOF
local cmp = require "cmp"
cmp.setup {
  snippet = {
    expand = function(args)
      vim.fn["vsnip#anonymous"](args.body)
    end,
  },

  mapping = {
    ['<CR>'] = cmp.mapping.confirm({ select = true })
  },

  sources = cmp.config.sources({
    { name = "nvim_lsp" },
    { name = "buffer" },
  }),
}
EOF

lua << EOF
local capabilities = require('cmp_nvim_lsp').default_capabilities()

require'lspconfig'.cssls.setup {
  capabilities = capabilities,
}
require("rust-tools").setup({
	server = {
		capabilities = capabilities,
		on_attach = lsp_attach,
	}
})
EOF
```

### Description

- create a rust project 
cargo new cmptest
cd cmptest
cargo build
nvim  .
type assert and press Ctrl-Space to open the completion window
if the terminal size is less than certain amount the main completion window and details windows overlap each other and the main completion window is not readable
![image](https://github.com/hrsh7th/nvim-cmp/assets/46407603/fc6d8d56-8857-4a2c-9993-4eefa606c03d)
if you resize the terminal then it fixes 

### Steps to reproduce

- create a rust project 
cargo new cmptest
cd cmptest
cargo build
nvim  .
type assert and press Ctrl-Space to open the completion window
if the terminal size is less than certain amount the main completion window and details windows overlap each other and the main completion window is not readable

### Expected behavior

Expected : windows resize according to terminal size instead of overlapping

### Actual behavior

completion windows overlap and cannot be read
![image](https://github.com/hrsh7th/nvim-cmp/assets/46407603/cc08ddcb-71f4-4494-b1b3-8bfb8637cdb0)

### Additional context

I used both neovim with my own config and also lunarvim (community config) they both behave the same way for certain rust types. this is reported by many users. 
my neovim nvim-cmp config 
https://github.com/salimp2009/deskmach_config/blob/main/nvim/.config/nvim/lua/user/plugins/nvim-cmp.lua

**Comment by lqzzy** [comment]
I have the same problem

**Comment by HidemaruOwO** [comment]
I checked and this issue seems to show how to fix it
https://github.com/hrsh7th/nvim-cmp/issues/1154

**Comment by salimp2009** [comment]
> I researched and this issue seems to show how to fix it #1154

yeah but I dont think that issue is even closed . and not sure if you really want to remove the menu because it shows where the completion is coming from . 
I currently have this 
```lua
	vim_item.menu = ({
		nvim_lsp = "[LSP]",
		nvim_lua = "[NVLUA]",
		luasnip = "[SNPT]",
		buffer = "[BFR]",
		path = "[PATH]",
		cmp_tabnine = "[TBNI]",
		codeium = "[CDEM]",
		copilot = "[CPLT]",
		crates = "[CRAT]",
		emoji = "[EMOJ]",
	})[entry.source.name]
```
That looks like a temporary workaround
